### PR TITLE
docs: cover remaining 0.22 migration points and refresh stale pages

### PR DESCRIPTION
**Pull request by antimora**
### Checklist

- [ ] Confirmed that `cargo run-checks` has been executed (comment-only Rust changes; ran `cargo fmt --check`, `typos`, and `mdbook build`).
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

Follow-up to #5762.

### Changes

Cross-checked the migration guide against every commit between `v0.21.0` and `main` and filled the gaps:

- Migration guide: `burn-train` builder changes (`with_default_checkpointers`, boxed renderer, progress logger), `Metric::compute` and `Numeric` changes, AUROC constructors, custom renderer/event processor changes, `DistributedContext`/`ExecutionStrategy::ddp`, MSRV 1.95, opt-in `sqlite`, feature renames/removals, `extension`-gated `from_primitive`, `TensorData` `try_*` readback, NaN and empty-axis reduction semantics, negative dimensions, per-side `ConvOptions::padding`, burn-store/`ParamId`/`Kind` renames.
- Stale pages: ONNX import (`Model<Flex>`, `Tensor::<Flex, 4>`, `~0.21`), config `init::<Wgpu>`, tensor `to_vec` and `empty_like` rows, no-std versions, missing 0.22 nn modules, learner options table, sqlite note in dataset, `compute` in the `Metric` trait listing.
- Source docs: restore the note that the first tensor creation locks device settings (`configure()` then returns `AlreadyInitialized`); fix the `burn` feature list (`server` is `remote-server`; add `remote`, `extension`, `optim`, `rl`, `safetensors`, `pytorch`).

### Testing

`mdbook build` passes and all relative links/anchors in the migration page resolve. `typos` and `cargo fmt --check` are clean. Comment-only changes to `burn` and `burn-tensor`.

### CODE REVIEW

File: burn-book/src/building-blocks/backend.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ let settings = device.settings();
 
 Configure defaults before the first tensor operation on that device. Device settings are initialized
 once; creating the first tensor uses the backend's defaults if configuration has not happened
-earlier.
+earlier, and a later `configure()` call returns `DeviceError::AlreadyInitialized`.
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-book/src/building-blocks/backend.md]: Agreed. Fixed in d8bb4009e: the sentence now reads "the first read of the settings, including by tensor creation or `settings()`, locks them to the backend's defaults, and a later `configure()` call returns `DeviceError::AlreadyInitialized`."

File: burn-book/src/migrating-to-0.22.md [hunk 2]
```diff
@@ -196,6 +196,79 @@ scheduler records, so it does not resume the full training checkpoint. Start wit
...
+  no longer provide `SqliteDataset` or `HuggingfaceDatasetLoader`; enable `burn/sqlite` as well.
+  `sqlite-bundled` is now an alias for `sqlite`. The `SqliteDatasetError::Sql` and `Row` payloads
+  changed type because the storage is backed by Turso instead of `rusqlite`.
+- `server` is renamed to `remote-server`. `remote` enables remote devices over Iroh, and
+  `remote-websocket` adds the WebSocket transport. The `router`, `dispatch`,
+  `record-item-custom-serde`, and `candle*` features are removed. New features include
+  `safetensors` and `pytorch` (which imply `store`), `linalg`, `signal`, `extension`, `capture`,
+  `optim`, and `rayon`.
+- `Tensor::from_primitive` and `into_primitive` require the `extension` feature and are generic
+  over the backend, for example `Tensor::from_primitive::<B>(primitive)`.
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-book/src/migrating-to-0.22.md]: Agreed. Fixed in d8bb4009e: the bullet now says `into_primitive` is replaced by `try_into_primitive::<B>()`, which returns an error when the tensor is not on backend `B`. The stale `tensor.into_primitive()` row in tensor.md was updated as well.

File: burn-book/src/migrating-to-0.22.md [hunk 3]
```diff
@@ -196,6 +196,79 @@ scheduler records, so it does not resume the full training checkpoint. Start wit
...
+
+## Tensor data and numeric semantics
+
+`TensorData::to_vec` and `into_vec` are deprecated in favor of `try_to_vec` and `try_into_vec`,
+which return the stored dtype without conversion. Use `try_to_vec_as::<E>()` and
+`try_into_vec_as::<E>()` to convert to another element type, on both `TensorData` and `Tensor`.
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-book/src/migrating-to-0.22.md]: Agreed. Fixed in d275413fb: "which return `Result<Vec<E>, DataError>` and fail when `E` does not match the stored dtype", followed by the `*_as` variants for conversion.

File: crates/burn-tensor/src/device.rs [hunk 4]
```diff
@@ -930,8 +930,9 @@ impl Device {
     /// creation time.
     ///
     /// Settings can only be initialized once per device. Configure defaults before creating
-    /// tensors or initializing model parameters; otherwise, tensor creation uses the backend's
-    /// defaults.
+    /// tensors or initializing model parameters: the first tensor created on the device locks
+    /// the settings to the backend's defaults, and any later call returns
+    /// [`DeviceError::AlreadyInitialized`].
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-tensor/src/device.rs]: Agreed. Fixed in d8bb4009e: the doc now says the first read of the settings locks them, and names both tensor creation (even with an explicit dtype, since `TensorCreationOptions::resolve_dtype` reads settings unconditionally) and `Device::settings` as reads.

File: crates/burn-tensor/src/device.rs [hunk 5]
```diff
@@ -1467,7 +1468,8 @@ impl Devices {
     /// creation time.
     ///
     /// Settings can only be initialized once per device. Configure defaults before creating
-    /// tensors or initializing model parameters.
+    /// tensors or initializing model parameters; the first tensor created on a device locks its
+    /// settings.
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-tensor/src/device.rs]: Agreed. Fixed in d8bb4009e: `Devices::configure` now says the first read of a device's settings, including by tensor creation, locks them. 7e62d5d58 also notes that it stops at the first error and leaves earlier devices configured.

**Review by laggui — COMMENTED** [review]
Could we keep the main migration guide focused on common upgrade steps, simplify the required feature changes, and group specialized APIs (custom metrics, renderers, storage adapters, and backend extensions) into a skippable section with links?

Example structure:
```
Migrating to Burn 0.22
  ├── Cargo features and toolchain
  ├── Types and devices
  ├── Autodiff is runtime state
  ├── Moving tensors and switching module state
  ├── Datasets and dataloaders
  ├── Migrating checkpoints
  ├── Training with `burn-train`
  ├── Tensor data and numeric semantics
  └── Custom integrations
      ├── Modules and optimizers
      ├── Custom metrics
      ├── Renderers and event processors
      ├── Distributed training
      ├── Storage adapters and checkpointers
      └── Backend extensions
```

Apart from this cleanup, LGTM

**Comment by antimora** [comment]
Reorganized in 0419da207 to match the proposed structure:

```
Migrating to Burn 0.22
  ├── Cargo features and toolchain
  ├── Types and devices
  ├── Autodiff is runtime state
  ├── Moving tensors and switching module state
  ├── Datasets and dataloaders
  ├── Migrating checkpoints
  ├── Training with `burn-train`
  ├── Tensor data and numeric semantics
  └── Custom integrations
      ├── Modules and optimizers
      ├── Custom metrics
      ├── Renderers and event processors
      ├── Distributed training
      ├── Storage adapters and checkpointers
      └── Backend extensions
```

The feature changes are now a single 0.21 to 0.22 table at the top. The handwritten `Module` impl note, `ParamId`, the `Optimizer` trait, `with_custom_checkpointers`, the burn-store adapter API, the `extension` feature, and the `TensorKind`/`AutodiffTensor` changes all moved under Custom integrations, which opens with a line saying it can be skipped. "Other migration points" is gone; cosine annealing moved to Training, and linalg/signal to the feature table. I also added one sentence on the `QuantScheme` builder rename (`with_level`/`with_param` to `per_block`/`per_tensor`), which the old closing sentence hinted at without stating.

File: burn-book/src/migrating-to-0.22.md [hunk 6]
```diff
@@ -4,6 +4,27 @@ Burn 0.22 selects backends at runtime through `Device`. Tensor operations follow
...
+| `sqlite` through `dataset` or `train`                                         | `sqlite`, enabled explicitly (`sqlite-bundled` is an alias)            |
+| Linear algebra in `burn::tensor::linalg`                                      | `linalg` feature; `burn::linalg` extension traits                      |
+| Signal processing in `burn::tensor::signal`                                   | `signal` feature; `burn::signal` or `burn::tensor::signal`             |
+| `burn-store` dependency for SafeTensors and PyTorch                           | `safetensors` and `pytorch` features (imply `store`)                   |
+| `candle`, `candle-cuda`, `candle-metal`                                       | Removed                                                                |
+| `router`, `dispatch`, `distributed`, `collective`, `record-item-custom-serde` | Removed; runtime dispatch and `burn::tensor::distributed` are built in |
+
+Remote servers and captured-graph interpreters must register the FFT custom operations from
+`burn-signal`; see
+[Signal Processing Functions](./building-blocks/tensor.md#signal-processing-functions).
```
Comment (laggui) [inline burn-book/src/migrating-to-0.22.md]: Minor nit: this isn't so much a "cargo feature", but we could move it into the table cell for 0.22 e.g.

```
signal` feature; `burn::signal` or `burn::tensor::signal`. See [Signal Processing Functions](./building-blocks/tensor.md#signal-processing-functions)
```

**Review by laggui — APPROVED** [review]
Just a minor leftover, but I'll approve in advance

File: burn-book/src/migrating-to-0.22.md (hunk 6 above)
Comment (antimora) [reply burn-book/src/migrating-to-0.22.md]: Agreed. Fixed in 79fed0112: the link now lives in the `signal` row of the table as suggested, and the trailing paragraph is gone. The linked section already covers the `register_fft_ops` requirement for remote servers and capture, so nothing is lost.
